### PR TITLE
Package the tool using setup.py

### PR DESCRIPTION
**Pull request by rgayon**

### CODE REVIEW

File: docker_explorer/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/python
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply docker_explorer/__init__.py]: add copyright header to be consistent with the other Pyhton files?

File: docker_explorer/de.py [hunk 2]
```diff
@@ -210,5 +208,9 @@ def Main(self):
     else:
       raise ValueError('Unhandled command %s' % options.command)
 
-if __name__ == '__main__':
+def main():
+  """Main function"""
   DockerExplorer().Main()
+
+if __name__ == '__main__':
```
Comment (joachimmetz) [inline docker_explorer/de.py]: I would also recommend moving de.py into a scripts subdirectory e.g. https://github.com/libyal/winreg-kb

File: docker_explorer/lib/aufs.py [hunk 3]
```diff
@@ -20,7 +20,7 @@
 import json
 import os
 
-from lib.storage import Storage
+from docker_explorer.lib.storage import Storage
```
Comment (joachimmetz) [inline docker_explorer/lib/aufs.py]: not sure which style you want to stick to (maybe something for a follow up CL) but I would recommend importing at module level not at class level

```
from docker_explorer.lib import storage
```

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+
+de_description = 'Docker forensic analysis tool.'
+
+de_long_description = (
+    'docker-explorer is a tool to help forensic analysis of offline Docker '
+    'installations')
+
+setup(
+    name='docker-explorer',
```
Comment (joachimmetz) [inline setup.py]: doesn't the `-` cause issues?

https://www.python.org/dev/peps/pep-0008/#package-and-module-names

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    url='https://github.com/google/docker-explorer',
+    author='docker-explorer development team',
+    license='Apache License, Version 2.0',
+    packages=find_packages(exclude=['tests']),
+    classifiers=[
+        'Development Status :: 4 - Beta',
+        'Operating System :: OS Independent',
+        'Programming Language :: Python',
+    ],
+    entry_points={
```
Comment (joachimmetz) [inline setup.py]: Doesn't this automatically create de.py ? 

http://setuptools.readthedocs.io/en/latest/setuptools.html
http://setuptools.readthedocs.io/en/latest/setuptools.html#automatic-script-creation

Is that what you want? or do you want to just ship de.py as a script?
https://github.com/libyal/winreg-kb/blob/master/setup.py#L164

File: tests.py [hunk 6]
```diff
@@ -73,9 +74,9 @@ def testParseArguments(self):
     options = de_test_object.ParseArguments()
     usage_string = de_test_object._argument_parser.format_usage()
     expected_usage = (
-        'usage: {0} [-h] [-r DOCKER_DIRECTORY] '
-        '{{mount,list,history}} ...\n'.format(os.path.basename(__file__)))
-    self.assertEqual(expected_usage, usage_string)
+        r'usage: (setup.py|{0}) \[-h\] \[-r DOCKER_DIRECTORY\] '
+        '{{mount,list,history}} ...\n').format(os.path.basename(__file__))
+    self.assertRegexpMatches(usage_string, expected_usage)
```
Comment (joachimmetz) [inline tests.py]: This seems to be deprecated in Python 3 https://docs.python.org/3/library/unittest.html, if you're planning to support Python 3 maybe add an issue for this.

File: tests.py [hunk 7]
```diff
@@ -132,9 +133,10 @@ def testDetectStorage(self):
   def testGetContainerInfo(self):
```
Comment (joachimmetz) [inline tests.py]: personal preference suggestion: add a docstring so you can see this in the unittest output what is being tested

File: .gitignore [hunk 8]
```diff
@@ -1 +1,3 @@
 *.pyc
+dist/*
```
[the remark this answers is not in the document]
Comment (rgayon) [reply .gitignore]: done

File: docker_explorer/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rgayon) [reply docker_explorer/__init__.py]: done

File: docker_explorer/de.py [hunk 9]
```diff
@@ -210,5 +208,9 @@ def Main(self):
     else:
       raise ValueError('Unhandled command %s' % options.command)
 
-if __name__ == '__main__':
+def main():
```
[the remark this answers is not in the document]
Comment (rgayon) [reply docker_explorer/de.py]: Opened #7

File: docker_explorer/de.py [hunk 10]
```diff
@@ -210,5 +208,9 @@ def Main(self):
...
+  """Main function"""
   DockerExplorer().Main()
+
```
[the remark this answers is not in the document]
Comment (rgayon) [reply docker_explorer/de.py]: Ripped from dftimewolf =)

File: setup.py (hunk 4 above)
Comment (rgayon) [reply setup.py]: fixed

File: tests.py (hunk 6 above)
Comment (rgayon) [reply tests.py]: Added to  #7

File: tests.py (hunk 7 above)
Comment (rgayon) [reply tests.py]: Done

File: docker_explorer/de.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (rgayon) [reply docker_explorer/de.py]: Fixed here, but probably need to update the pylintrc in another PR

File: docker_explorer/de.py (hunk 2 above)
Comment (rgayon) [reply docker_explorer/de.py]: Keeping it as is, as I don't see any other use of the "docker_explorer" module than through the de.py file.

File: docker_explorer/lib/aufs.py (hunk 3 above)
Comment (rgayon) [reply docker_explorer/lib/aufs.py]: Done

**Comment by rgayon** [comment]
Thanks, good catch =)

**Pull request merged**